autoload/ledger.vim

```diff
@@ -400,3 +401,231 @@ func! ledger#entry()
+    return
+  endif
+  if empty(l:output)
+    call s:warning_message('No results')
+    return
+  endif
+  " Open a new buffer to show Ledger's output.
+  execute get(s:winpos_map, g:ledger_winpos, "bo new")
+  setlocal buftype=nofile bufhidden=wipe nobuflisted noswapfile nowrap
+  call append(0, l:output)
+  setlocal nomodifiable
+  " Set local mappings to quit window or lose focus.
+  nnoremap <silent> <buffer> <tab> <c-w><c-w>
+  nnoremap <silent> <buffer> q <c-w>c
+  " Add some coloring to the report
+  syntax match LedgerNumber /[^-]\d\+\([,.]\d\+\)\+/
+  syntax match LedgerNegativeNumber /-\d\+\([,.]\d\+\)\+/
+  syntax match LedgerImproperPerc /\d\d\d\+%/
+endf
+
+" Show a register report in a quickfix list.
+function! ledger#register(args)
+  if !s:is_ledger_buffer() | return | endif
+  let l:cmd = s:ledger_cmd(extend([
+        \ "register",
+        \ "-f", "%",
+        \ "--format='" . g:ledger_qf_register_format . "'",
+        \ "--prepend-format='%(filename):%(beg_line) '"
+        \ ], split(a:args, ' ')))
+  if g:ledger_debug | return l:cmd | endif
+  call s:quickfix_populate(systemlist(l:cmd))
+  call s:quickfix_toggle('Register report')
+endf
+
+function! ledger#reconcile(account, target_amount)
+  if !s:is_ledger_buffer() | return | endif
+  let l:cmd = s:ledger_cmd([
+        \ "register",
+        \ a:account,
+        \ "--uncleared",
+        \ "--format='" . g:ledger_qf_reconcile_format . "'",
+        \ "--prepend-format='%(filename):%(beg_line) %(pending ? \"P\" : \"U\") '"
+        \ ])
+  if g:ledger_debug | return l:cmd | endif
+  call s:quickfix_populate(systemlist(l:cmd))
+  if s:quickfix_toggle("Reconcile '" . a:account . "' account", "Nothing to reconcile")
+    let g:ledger_target_amount = a:target_amount
+    " Show updated account balance upon saving, as long as the quickfix window is open
+    augroup reconcile
+      autocmd!
+      execute "autocmd BufWritePost *.ldg,*.ledger call ledger#show_balance('" . a:account . "')"
+      autocmd BufWipeout <buffer> call <sid>finish_reconciling()
+    augroup END
+    call ledger#show_balance(a:account)
+  endif
+endf
+
+function! s:finish_reconciling()
+  unlet g:ledger_target_amount
+  augroup reconcile
+    autocmd!
+  augroup END
+  augroup! reconcile
+endf
+
+" Show the pending/cleared balance of a given account,
+" or of the account in the current line if no argument is given.
+function! ledger#show_balance(...)
+  let l:account = a:0 > 0 && !empty(a:1) ? a:1 : matchstr(getline('.'), '\m\(  \|\t\)\zs\S.\{-}\ze\(  \|\t\|$\)')
+  if empty(l:account)
+    call s:error_message('No account found')
+    return
+  endif
+  let l:cmd = s:ledger_cmd([
+        \ 'cleared',
+        \ l:account,
+        \ "--empty",
+        \ "--collapse",
+        \ "--format='%(scrub(get_at(display_total, 0)))|%(scrub(get_at(display_total, 1)))|%(quantity(scrub(get_at(display_total, 1))))'",
+        \ (exists("g:ledger_default_commodity") ? "-X " . g:ledger_default_commodity : '')
```

Comment:
We should check if `g:ledger_default_commodity` isn't empty instead of / in addition to the `exists` check. Its existence is ensured in `ftplugin/ledger.vim` anyways.

With the default value ledger complains: `No argument provided for --exchange`.